### PR TITLE
Fully deprecate variadic inputs of checkpoint_sequential

### PR DESCRIPTION
**Pull request by sublee**
To support variadic inputs of `checkpoint_sequential` was deprecated at #21006. This case should be warned with `DeprecationWarning` for PyTorch 1.2, but it should be simply failed with `TypeError` since PyTorch 1.3. This patch removes the `DeprecationWarning` for PyTorch 1.2.

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by sublee** [comment]
@soumith Thank you for reviewing it.

**Comment by yf225** [comment]
@sublee We can't land it yet because test is failing:
```
Sep 24 20:52:17 ======================================================================
Sep 24 20:52:17 ERROR: test_checkpoint_module_list_multiple_args (__main__.TestCheckpoint)
Sep 24 20:52:17 ----------------------------------------------------------------------
Sep 24 20:52:17 Traceback (most recent call last):
Sep 24 20:52:17   File "test_utils.py", line 188, in test_checkpoint_module_list_multiple_args
Sep 24 20:52:17     torch.randn(1, 60, requires_grad=True)
Sep 24 20:52:17   File "test_utils.py", line 69, in _check_checkpoint_sequential
Sep 24 20:52:17     out = checkpoint_sequential(model_to_compare, num_chunks, *detached_inputs)
Sep 24 20:52:17   File "/opt/conda/lib/python3.6/site-packages/torch/utils/checkpoint.py", line 217, in checkpoint_sequential
Sep 24 20:52:17     preserve_rng_state=preserve_rng_state)
Sep 24 20:52:17   File "/opt/conda/lib/python3.6/site-packages/torch/utils/checkpoint.py", line 155, in checkpoint
Sep 24 20:52:17     return CheckpointFunction.apply(function, preserve, *args)
Sep 24 20:52:17   File "/opt/conda/lib/python3.6/site-packages/torch/utils/checkpoint.py", line 62, in forward
Sep 24 20:52:17     if preserve_rng_state:
Sep 24 20:52:17 RuntimeError: bool value of Tensor with more than one value is ambiguous
```

**Review by soumith — CHANGES_REQUESTED** [review]
please fix test

**Comment by sublee** [comment]
@yf225, @soumith Sorry for my mistake. I ran only the test cases I added at #21006 but I had not to do. I just fixed those failures. Please review it.

**Comment by sublee** [comment]
@soumith I guess you lost the last update. Would you please review this PR again?

**Comment by sublee** [comment]
This patch was intended to deploy with v1.3.0. However, today PyTorch v1.3.0 has been released. I update the version to v1.4 in the deprecation documentation and rebase this branch onto the current `master`.

File: test/test_utils.py [hunk 1]
```diff
@@ -44,41 +44,37 @@ def _check_checkpoint_sequential(
...
+        out = model(input)
         out_not_checkpointed = out.data.clone()
         model.zero_grad()
         out.sum().backward()
         grad_not_checkpointed = {
             name: param.grad.data.clone()
             for name, param in model.named_parameters()
         }
-        input_grad_not_checkpointed = [i.grad.data.clone() for i in inputs]
+        input_grad_not_checkpointed = input.grad.data.clone()
```
Comment (albanD) [inline test/test_utils.py]: Can you use `.detach()` here instead of `.data` please?

File: test/test_utils.py [hunk 2]
```diff
@@ -44,41 +44,37 @@ def _check_checkpoint_sequential(
...
+            out = checkpoint_sequential(model_to_compare, num_chunks, detached)
             out_checkpointed = out.data.clone()
             model.zero_grad()
             out.sum().backward()
             grad_checkpointed = {
                 name: param.grad.data.clone()
                 for name, param in model.named_parameters()
             }
-            input_grad_checkpointed = [d.grad.data.clone() for d in detached_inputs]
+            input_grad_checkpointed = detached.grad.data.clone()
```
Comment (albanD) [inline test/test_utils.py]: Same data -> detach()

**Review by albanD — APPROVED** [review]
Details in the test to make it look slightly better.
Then I'll merge.

Thanks for the PR !

File: test/test_utils.py (hunk 1 above)
Comment (sublee) [reply test/test_utils.py]: Sure! I agree with you.

File: test/test_utils.py (hunk 2 above)
Comment (sublee) [reply test/test_utils.py]: I applied your suggestion: https://github.com/pytorch/pytorch/pull/25985/commits/01bf4641093a9396aac90976ee4bd4c215d412cc

**Comment by albanD** [comment]
Just rebased on top of master for CI.

**Review by soumith — APPROVED** [review]

**Comment by zou3519** [comment]
Hmm, If I am understanding this correctly it is not BC breaking because it enables behavior that used to throw a TypeError

**Pull request closed without merging**